service.rb

```diff
@@ -7,49 +7,42 @@ class Service
   def initialize(port)
     app = -> environment {
       request = Rack::Request.new(environment)
+      response = Rack::Response.new
       store = YAML::Store.new("daily_data.yml")
 
       if request.get? && request.path == "/show/data"
-        content_type = "text/html"
-        status = 200
-        response_message = "<ul>\n"
+        response.write("<ul>\n")
+        response.content_type = "text/html; charset=UTF-8"
 
         all_data = {}
         store.transaction do
           all_data = store[:all_data]
         end
 
         all_data.each do |daily_data|
-          response_message << "<li> On this day <b>#{CGI.escapeHTML(daily_data[:date])}</b>, #{CGI.escapeHTML(daily_data[:step_count])}, #{CGI.escapeHTML(daily_data[:notes])}</li>\n"
+          response.write "<li> On this day <b>#{CGI.escapeHTML(daily_data[:date])}</b>, #{CGI.escapeHTML(daily_data[:step_count])}, #{CGI.escapeHTML(daily_data[:notes])}</li>\n"
         end
 
-        response_message << "</ul>\n"
-        response_message << daily_data_form
+        response.write "</ul>\n"
+        response.write daily_data_form
 
       elsif request.post? && request.path == "/add/data"
-        content_type = "text/html"
-        status = 303
-        response_message = ""
+        response.write("")
+        response.content_type = "text/html; charset=UTF-8"
```

Comment:
I should’ve spotted this [sooner](https://github.com/wildmaples/mirth/commit/d9030c2d1aa3d953100fdbf48ce33257098c9c54#diff-9454335d29f10b034af0bfd6d2b278a3e099f78eb23816964243f9eaa27463daL56), but we’re no longer sending a response body here so the `Content-Type` doesn’t matter. You can just remove this line.